### PR TITLE
Fix flaky spark session creation

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

## What changes are proposed in this pull request?

A tempoarary workaround for #4229

I'm investigating why this issue suddenly started happening.

## How is this patch tested?

(Details)

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [x] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by harupy** [comment]
https://github.com/dotnet/spark/issues/435

File: tests/pyfunc/test_spark.py [hunk 1]
```diff
@@ -60,11 +60,20 @@ def get_spark_session(conf):
...
+    MAX_ATTEMPTS = 5
+    for attempt in range(MAX_ATTEMPTS):
+        try:
+            return (
+                pyspark.sql.SparkSession.builder.config(conf=conf)
+                .master("local-cluster[2, 1, 1024]")
+                .getOrCreate()
+            )
+        except Exception as e:
+            print("Attempt {} / {} faild: {}".format(attempt, MAX_ATTEMPTS, e))
```
Comment (harupy) [inline tests/pyfunc/test_spark.py:74]: I've tried:

```
conf.set("spark.driver.bindAddress","127.0.0.1")
```

but it didn't work.

Comment (harupy) [reply tests/pyfunc/test_spark.py:74]: I think we need to do the same thing in java tests

**Review by WeichenXu123 — APPROVED** [review]

**Comment by WeichenXu123** [comment]
Withdraw approval above.
I guess this may due to old spark sessions not closed. And you create new one and exhaust port space.
Maybe other processes did not quit properly so keep sessions not closed.
try `ps | grep spark` to check whether other processes exit correctly
The pytest command will spawn multiple processes to run tests

**Comment by harupy** [comment]
@WeichenXu123 Thanks for the comment. This issue occurs in the first attempt of spark session creation:
https://github.com/mlflow/mlflow/runs/2284103398#step:5:226

File: dev/run-python-sagemaker-tests.sh [hunk 2]
```diff
@@ -20,6 +20,7 @@ pytest --verbose tests/sagemaker/mock --large
 
 # Added here due to dependency on sagemaker
 # TODO: split out sagemaker tests and move other spark tests to run-python-flavor-tests.sh
+ps | grep spark
```
Comment (harupy) [inline dev/run-python-sagemaker-tests.sh:23]: Revert this change
